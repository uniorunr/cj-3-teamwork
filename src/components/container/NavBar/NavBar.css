.navbar-container {
  width: 100%;
  height: 10vh;
  border-bottom: 1px solid #000;
}

.navbar-container nav {
  height: 100%;
}

.navbar-container nav ul {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}
