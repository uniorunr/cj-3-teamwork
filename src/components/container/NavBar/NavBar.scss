$navBarColor: #fff;

.navbar-container {
  position: fixed;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 10vh;
  background: linear-gradient(to bottom, rgba(0, 0, 0, 1) 0%, rgba(0, 0, 0, 0.5) 50%, rgba(0, 0, 0, 0) 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#ffffff', endColorstr='#00ffffff',GradientType=0 );

  .header-logo {
    padding: 0 25px;

    h1 {
      color: $navBarColor;
      text-shadow: 0 1px 2px #000, 0 2px 5px #000, 0 3px 10px #000;
    }
  }

  nav {
    height: 100%;

    ul {
      display: flex;
      height: 100%;
    }
  }
}
