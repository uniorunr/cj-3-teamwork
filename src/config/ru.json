{
  "navbar": {
    "home": "главная",
    "list": "список режиссеров"
  },
  "main": {
    "content": {
      "name": "Александровская Лариса Помпеевна",
      "description": "Белорусская советская оперная певица (сопрано), режиссёр, публицист и общественный деятель. Народная артистка СССР (1940)",
      "foto": "https://upload.wikimedia.org/wikipedia/commons/0/0c/%C5%81arysa_Aleksandro%C5%ADskaja._%D0%9B%D0%B0%D1%80%D1%8B%D1%81%D0%B0_%D0%90%D0%BB%D0%B5%D0%BA%D1%81%D0%B0%D0%BD%D0%B4%D1%80%D0%BE%D1%9E%D1%81%D0%BA%D0%B0%D1%8F_%281937%29.jpg",
      "title": "Что такое Lorem Ipsum?",
      "text": "Lorem Ipsum - это текст-\"рыба\", часто используемый в печати и вэб-дизайне. Lorem Ipsum является стандартной \"рыбой\" для текстов на латинице с начала XVI века. В то время некий безымянный печатник создал большую коллекцию размеров и форм шрифтов, используя Lorem Ipsum для распечатки образцов. Lorem Ipsum не только успешно пережил без заметных изменений пять веков, но и перешагнул в электронный дизайн. Его популяризации в новое время послужили публикация листов Letraset с образцами Lorem Ipsum в 60-х годах и, в более недавнее время, программы электронной вёрстки типа Aldus PageMaker, в шаблонах которых используется Lorem Ipsum."
    }
  }
}
