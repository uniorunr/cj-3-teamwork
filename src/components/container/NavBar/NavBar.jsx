import React from 'react';
import NavBarItem from '../../presentetion/NavBarItem/NavBarItem';
import './NavBar.scss';

const NavBar = () => (
  <div className="navbar-container">
    <div className="header-logo">
      <h1>Director</h1>
    </div>
    <nav>
      <ul>
        <NavBarItem hash="/" name="Home" />
        <NavBarItem hash="/about" name="About" />
        <NavBarItem hash="/users" name="Users" />
      </ul>
    </nav>
  </div>
);

export default NavBar;
