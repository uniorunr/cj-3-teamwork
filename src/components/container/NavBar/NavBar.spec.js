import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';
import ru from '../../../config/ru.json';

it('renders correctly', () => {
  const tree = renderer.create(<NavBar lang={ru} active onClick={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
