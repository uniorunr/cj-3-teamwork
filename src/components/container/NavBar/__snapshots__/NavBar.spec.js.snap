// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar-container"
>
  <div
    className="header-logo"
  >
    <h1>
      Director
    </h1>
  </div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
